### PR TITLE
Add RPC errors to some VTGate calls

### PR DESCRIPTION
**Pull request by aaijazi**
@guoliang100 @alainjobart 

I added an RPCError field to all VTGate non-batch Execute* calls, and polished up tests etc. so that they should all be tested and working. Each change is in a small, self-contained commit for easy reviewing.

Next, I'm going to do the same for the other non-streaming calls, and finally I'll change the streaming calls. @guoliang100, we should sync up about what changes I should make to Begin/Commit/Rollback.

### CODE REVIEW

File: go/vt/vtgate/vtgate.go [hunk 1]
```diff
@@ -84,6 +85,11 @@ type RegisterVTGate func(vtgateservice.VTGateService)
 // RegisterVTGates stores register funcs for VTGate server.
 var RegisterVTGates []RegisterVTGate
 
+var (
+	// RPCErrorOnlyInReply informs vtgateservice(s) about how to return errors.
+	RPCErrorOnlyInReply = flag.Bool("rpc-error-only-in-reply", false, "if true, supported RPC calls from vtgateservice(s) will only return errors as part of the RPC server response")
```
Comment (guoliang100) [inline go/vt/vtgate/vtgate.go]: Please sync with Jeff, as he thinks we should pass the error back as return value as well.

**Comment by guoliang100** [comment]
LGTM

**Comment by aaijazi** [comment]
@guoliang100 

As we discussed today, I added Begin2/Commit2/Rollback2. 

This should be all the non-streaming calls, with solid tests.

If this looks good to you, I'm going to import into Google3, and then try to get the streaming calls in the next pull request.

**Comment by guoliang100** [comment]
LGTM

**Pull request merged**